src/components/TextInput/index.test.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+import { render, screen } from '@testing-library/react';
+
+import TextInput from '.';
+
+describe('TextInput', () => {
+  const label = 'Email';
+  const labelDataTestId = 'email-label';
+  const inputDataTestId = 'text-input';
+  const inputType = 'email';
+  const inputPlaceholder = 'input placeholder';
+  it('renders a text input with a label', () => {
+    const inputProps = {
+      label: label,
+      labelDataTestId: labelDataTestId,
+      inputAttributes: {
+        id: 'sign-in-email',
+        required: true,
+        type: inputType,
+        placeholder: inputPlaceholder,
+        'data-test-id': inputDataTestId,
+      },
+    };
+    render(<TextInput {...inputProps} />);
+
+    const inputLabel = screen.getByTestId(labelDataTestId);
+    const textInput = screen.getByTestId(inputDataTestId);
+
+    expect(inputLabel).toBeVisible();
+    expect(inputLabel).toHaveTextContent(label);
+
+    expect(textInput).toBeVisible();
+    expect(textInput).toHaveAttribute('type', inputType);
+    expect(textInput).toHaveAttribute('placeholder', inputPlaceholder);
+  });
+
+  it('renders a text input without a label', () => {
+    const inputProps = {
+      inputAttributes: {
+        id: 'sign-in-email',
+        required: true,
+        type: inputType,
+        placeholder: inputPlaceholder,
+        'data-test-id': inputDataTestId,
+      },
+    };
+    render(<TextInput {...inputProps} />);
+
+    expect(screen.queryByRole('label')).not.toBeInTheDocument();
```

Comment:
(toBeInTheDocument)[https://github.com/testing-library/jest-dom#tobeinthedocument] passes if the element is `display: none` or `visibility: hidden`, which we don't want in this case, so could we use `toBeVisible`?

```suggestion
    expect(screen.queryByRole('label')).not.toBeVisible();
```

Reply:
@tyrro actually when the label is empty, the `<label>` element will not be generated so `toBeInTheDocument()` is my expected result in this case 😄 

I also try to use `toBeVisible()` but it doesn't work with the does-not-exist element 🙏 
![image](https://github.com/manh-t/react-survey/assets/60863885/90ae4a62-9d0d-458a-aa8e-6465cf965aeb)

Reply:
ah I see, we are testing a non-existing element!